src/executables/util_service.c

```diff
@@ -2279,6 +2344,241 @@ process_manager (int command_type, bool process_window_service)
+    }
+}
+
+static int
+process_javasp_start (const char *db_name, bool process_window_service)
+{
+  static const int wait_timeout = 30;
+  int waited_secs = 0;
+  int pid = 0;
+  int status = ER_GENERIC_ERROR;
+
+  print_message (stdout, MSGCAT_UTIL_GENERIC_START_STOP_3S, PRINT_JAVASP_NAME, PRINT_CMD_START, db_name);
+
+  UTIL_JAVASP_SERVER_STATUS_E javasp_status = is_javasp_running (db_name);
+  if (javasp_status == JAVASP_SERVER_RUNNING)
+    {
+      print_message (stdout, MSGCAT_UTIL_GENERIC_ALREADY_RUNNING_2S, PRINT_JAVASP_NAME, db_name);
+      util_log_write_errid (MSGCAT_UTIL_GENERIC_ALREADY_RUNNING_2S, PRINT_JAVASP_NAME, db_name);
+    }
+  else if (javasp_status == JAVASP_SERVER_STOPPED)
+    {
+      if (process_window_service)
+	{
+#if defined(WINDOWS)
+	  const char *args[] = { UTIL_WIN_SERVICE_CONTROLLER_NAME, PRINT_CMD_JAVASP,
+	    COMMAND_TYPE_START, db_name, NULL
+	  };
+	  status = proc_execute (UTIL_WIN_SERVICE_CONTROLLER_NAME, args, true, false, false, NULL);
+#endif
+	}
+      else
+	{
+	  const char *args[] = { UTIL_JAVASP_NAME, db_name, NULL };
+	  status = proc_execute (UTIL_JAVASP_NAME, args, false, false, false, &pid);
+	}
+
+      status = ER_GENERIC_ERROR;
+      while (status != NO_ERROR && waited_secs < wait_timeout)
+	{
+	  if (pid != 0 && is_terminated_process (pid))
+	    {
+	      break;
+	    }
+
+	  javasp_status = is_javasp_running (db_name);
+	  if (javasp_status == JAVASP_SERVER_RUNNING)
+	    {
+	      status = NO_ERROR;
+	      break;
+	    }
+	  else if (javasp_status == JAVASP_SERVER_STOPPED)
+	    {
+	      sleep (1);	/* wait to start */
+	      waited_secs++;
+
+	      util_log_write_errstr ("Waiting for javasp server to start... (%d / %d)\n", waited_secs, wait_timeout);
+	    }
+	  else
+	    {
+	      /* invalid database name or failed to open info file */
+	      break;
+	    }
+	}
+    }
+
+  print_result (PRINT_JAVASP_NAME, status, START);
+  return status;
+}
+
+static int
+process_javasp_stop (const char *db_name, bool process_window_service)
+{
+  int status = NO_ERROR;
+
+  print_message (stdout, MSGCAT_UTIL_GENERIC_START_STOP_3S, PRINT_JAVASP_NAME, PRINT_CMD_STOP, db_name);
+  if (is_javasp_running (db_name) == JAVASP_SERVER_RUNNING)
+    {
+      if (process_window_service)
+	{
+#if defined(WINDOWS)
+	  const char *args[] = { UTIL_WIN_SERVICE_CONTROLLER_NAME, PRINT_CMD_JAVASP,
+	    COMMAND_TYPE_STOP, db_name, NULL
+	  };
+
+	  status = proc_execute (UTIL_WIN_SERVICE_CONTROLLER_NAME, args, true, false, false, NULL);
+	  status = (is_javasp_running (db_name) != JAVASP_SERVER_STOPPED) ? ER_GENERIC_ERROR : NO_ERROR;
+#endif
+	}
+      else
+	{
+	  const char *args[] = { UTIL_JAVASP_NAME, COMMAND_TYPE_STOP, db_name, NULL };
+	  status = proc_execute (UTIL_JAVASP_NAME, args, true, false, false, NULL);
+	  if (status == NO_ERROR)
+	    {
+	      sleep (1);	/* wait to stop */
+	      status = (is_javasp_running (db_name) != JAVASP_SERVER_STOPPED) ? ER_GENERIC_ERROR : NO_ERROR;
```

Comment:
How about like this?
```
while (1)
{
     sleep (1);	/* wait to stop */
     status = (is_javasp_running (db_name) != JAVASP_SERVER_STOPPED) ? ER_GENERIC_ERROR : NO_ERROR;
     if (status == JAVASP_SERVER_STOPPED || cnt++ >5) break;
}
```

Reply:
I modified as below. It is the same pattern with process_javasp_start ()
```
  static const int wait_timeout = 5;
  int waited_secs = 0;
	  while (status == NO_ERROR && waited_secs < wait_timeout)
	    {
	      sleep (1);	/* wait to stop */
	      status = (is_javasp_running (db_name) != JAVASP_SERVER_STOPPED) ? ER_GENERIC_ERROR : NO_ERROR;
        waited_secs++;
	    }
```